specifyweb/frontend/js_src/lib/components/Formatters/Formatter.tsx

```diff
@@ -116,11 +168,28 @@ function Definitions({
                     : resourcesText.conditionDescription()}
                 </span>
               </Label.Block>
-            )}
+              <Fields
+                fields={[
+                  fields,
+                  (fields): void => handleChanged({ value, fields }, index),
+                ]}
+                table={table}
+                onDelete={
+                  trimmedFields.length < 2
+                    ? undefined
+                    : (): void =>
+                        handleChange(
+                          removeItem(formatter.definition.fields, index)
+                        )
+                }
+              />
+            </>
+          )}
```

Comment:
a lot of code duplication here. could you please extract this into a variable and reuse it?

but really, even that shouldn't be needed. could be simplified to look like this:

```tsx
.map((...)=>{
  <div ...>
    {hasCondition && <Label.Block>...</Label.Block>}
    <Fields ... />
  </div>
})
```